packages/core/src/test/shared/ui/sam/stackPrompter.test.ts

```diff
@@ -0,0 +1,132 @@
+import * as SamUtilsModule from '../../../../shared/sam/utils'
+import * as ButtonsModule from '../../../../shared/ui/buttons'
+import { DefaultCloudFormationClient } from '../../../../shared/clients/cloudFormationClient'
+import { samSyncUrl } from '../../../../shared/constants'
+import { createStackPrompter } from '../../../../shared/ui/sam/stackPrompter'
+import { intoCollection } from '../../../../shared/utilities/collectionUtils'
+import { sleep } from '../../../../shared/utilities/timeoutUtils'
+
+describe('createStackPrompter', () => {
+    let sandbox: sinon.SinonSandbox
+    const cfnClient = new DefaultCloudFormationClient('us-east-1')
+    const mementoRootKey = 'samcli.sync.params'
+
+    beforeEach(() => {
+        sandbox = sinon.createSandbox()
+    })
+
+    afterEach(() => {
+        sandbox.restore()
+    })
+
+    it('should create a prompter with existing stacks', async () => {
+        // Arrange
+        const stackSummaries: CloudFormation.StackSummary[][] = [
+            [
+                {
+                    StackName: 'stack1',
+                    StackStatus: 'CREATE_COMPLETE',
+                    CreationTime: new Date(),
+                } as CloudFormation.StackSummary,
+                {
+                    StackName: 'stack2',
+                    StackStatus: 'CREATE_COMPLETE',
+                    CreationTime: new Date(),
+                } as CloudFormation.StackSummary,
+                {
+                    StackName: 'stack3',
+                    StackStatus: 'CREATE_COMPLETE',
+                    CreationTime: new Date(),
+                } as CloudFormation.StackSummary,
+            ],
+        ]
+        const expectedItems = [
+            {
+                label: 'stack1',
+                data: 'stack1',
+                description: undefined,
+                invalidSelection: false,
+                recentlyUsed: false,
+            },
+            {
+                label: 'stack2',
+                data: 'stack2',
+                description: undefined,
+                invalidSelection: false,
+                recentlyUsed: false,
+            },
+            {
+                label: 'stack3',
+                data: 'stack3',
+                description: undefined,
+                invalidSelection: false,
+                recentlyUsed: false,
+            },
+        ]
+        const listAllStacksStub = sandbox.stub(cfnClient, 'listAllStacks').returns(intoCollection(stackSummaries))
+        sandbox.stub(SamUtilsModule, 'getRecentResponse').returns(undefined)
+        const createCommonButtonsStub = sandbox.stub(ButtonsModule, 'createCommonButtons')
+        sandbox
+            .stub(AwsConsoleModule, 'getAwsConsoleUrl')
+            .returns(vscode.Uri.parse(`https://us-east-1.console.aws.amazon.com/cloudformation/home?region=us-east-1`))
+
+        // Act
+        const prompter = createStackPrompter(cfnClient, mementoRootKey, samSyncUrl)
+        await sleep(50)
```

Comment:
Is this to give the prompter time to be clicked asynchronously?
If so, I think this might be susceptible to flakiness if the machine can't do it in 50ms.

Why cant we do something like
```suggestion
        await waitUntil(async () => return createCommonButtonStub.calledOnce, { timeout: 3000, interval: 100 })
```

Reply:
That's a possibility. I have attempted to use PromperTester class to test the actual prompt. The issue seems to be the stackPrompter or bucketPrompter need some moments to render item from list api. Calling `prompt()` right after calling  create prompter method will result in empty option items. Even using `await picker.untilReady()` does not solve this problem at the moment. 

However, the issue only exists when te call prompter directly. The in sync/deploy wizard test, the prompters seem to work correctly; because of this, I am relax about this test cases.

Reply:
Give suggested solution a try and did not work.